executor/compact_table.go

```diff
@@ -158,7 +159,27 @@ func (task *storeCompactTask) work() error {
 				// Stop remaining partitions when error happens.
 				break
 			}
-		} // For partition table, there must be no data in task.parentExec.tableInfo.ID. So no need to compact it.
+		}
+	} else if task.parentExec.PartitionNames != nil {
+		// User specify partitions, let's do it partition by partition.
+		// There is no need for partition-level concurrency, as TiFlash will limit table compaction one at a time.
+		allPartitions := task.parentExec.tableInfo.Partition.Definitions
+		task.allPhysicalTables = len(allPartitions)
```

Comment:
We should assign the partition to be compacted as `allPhysicalTables`, otherwise the progress report in the log will be incorrect.